macros/utils/table_operations/replace_data.sql

```diff
@@ -0,0 +1,43 @@
+{% macro replace_data(relation, sql_query) %}
+    {{ return(adapter.dispatch('replace_data', 'elementary')(relation, sql_query)) }}
+{% endmacro %}
+
+{% macro replace_data_with_table_contents(relation, replacing_relation) %}
+    {% do return(elementary.replace_data(relation, 'select * from {}'.format(replacing_relation))) %}
+{% endmacro %}
+
+{# Databricks / Spark (non-atomic implementation) #}
+{% macro default__replace_data(relation, sql_query) %}
+    {% do dbt.truncate_relation(relation) %}
+    {% do dbt.run_query(elementary.insert_as_select(relation, sql_query)) %}
+{% endmacro %}
+
+{% macro snowflake__replace_data(relation, sql_query) %}
```

Comment:
Not sure if it's critical, but technically in both Snowflake and BQ there's a MERGE statement - that may allow doing this without a transaction.
I thought maybe this is simpler and more clear though.